src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -207,12 +207,41 @@ def build_summary_data(
                 group__in=([group for group in regressed_or_escalated_groups]),
                 type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_ESCALATING.value),
             )
-            if regressed_or_escalated_groups_today:
-                for activity in regressed_or_escalated_groups_today[:4]:
-                    if activity.type == ActivityType.SET_REGRESSION.value:
-                        project_ctx.regressed_today.append(activity.group)
-                    else:
-                        project_ctx.escalated_today.append(activity.group)
+
+            deduped_groups_by_activity_type: dict[int, set] = {
+                ActivityType.SET_REGRESSION.value: set(),
+                ActivityType.SET_ESCALATING.value: set(),
```

Comment:
Nit - prefer using the enum member instead of the value, this allows us to have a stricter (more useful) type on the dict.
```suggestion
            deduped_groups_by_activity_type: dict[ActivityType, set] = {
                ActivityType.SET_REGRESSION: set(),
                ActivityType.SET_ESCALATING: set(),
```